From 3cc0adf9a95e02999e3005b6dbcc2609e332ae6b Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 23 May 2017 18:01:15 +0800
Subject: [PATCH 1409/5242] MLK-14966-1 soc: imx: add ARCH_MXC_ARM64
 menuconfig

commit  cc14c9172ed52946a82beaf84f1f247b70524f55 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add ARCH_MXC_ARM64 for i.MX ARM64 platforms, this
can improve the modules dependency logic.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/Kconfig |   25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/drivers/soc/imx/Kconfig b/drivers/soc/imx/Kconfig
index a5b86a2..68cbe4c 100644
--- a/drivers/soc/imx/Kconfig
+++ b/drivers/soc/imx/Kconfig
@@ -1,5 +1,30 @@
 menu "i.MX SoC drivers"
 
+menuconfig ARCH_MXC_ARM64
+	bool "NXP i.MX family"
+	depends on ARM64
+	help
+	  Support for NXP MXC/iMX-based ARM64 family of processors
+
+if ARCH_MXC_ARM64
+
+config ARCH_FSL_IMX8QM
+	bool "i.MX8QM"
+	select HAVE_IMX_MU
+	select HAVE_IMX_RPMSG
+
+config ARCH_FSL_IMX8QXP
+	bool "i.MX8QXP"
+	select HAVE_IMX_MU
+	select HAVE_IMX_RPMSG
+
+config ARCH_FSL_IMX8MQ
+	bool "i.MX8MQ"
+	select HAVE_IMX_MU
+	select HAVE_IMX_RPMSG
+
+endif
+
 config IMX7_PM_DOMAINS
 	bool "i.MX7 PM domains"
 	depends on SOC_IMX7D || (COMPILE_TEST && OF)
-- 
1.7.9.5

